Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -469,7 +482,15 @@ public final class PackageBuilder {
                 if subpath == "" || subpath == "." {
                     return packagePath
                 }
-                let path = packagePath.appending(RelativePath(subpath))
+				
+				// Make sure the target path doesn't contain '~'.
+				guard subpath.first != "~" else {
+					throw ModuleError.unsupportedPath(path: subpath, reason: .unsupportedTilda)
+				}
+				
+				let path = subpath.first == "/" ?
+					AbsolutePath(subpath) : packagePath.appending(RelativePath(subpath))
```

Comment:
Instead of these checks here, we should add a new static method to `RelativePath` to validate a given input string. We can do something like this:

```swift
struct RelativePath {
    ...
    static func isValid(path: String) -> Bool
    ...
}
```

Reply:
Found that 
``` swift
let path = AbsolutePath(subpath, relativeTo: packagePath)
```
does exactly what needed!